src/DynamoCore/Models/PortModel.cs

```diff
@@ -91,7 +91,7 @@ public bool IsConnected
             }
         }
 
-        public string ToolTipContent { get; private set; }
+        public string ToolTipContent { get; set; }
```

Comment:
@pboyer , the context is, if the port has been created, we only need to update it instead of creating a new one:
```
public PortModel AddPort(PortType portType, PortData data, int index)
{
    ...
    if (inPorts.Count > index)
    {
        p = inPorts[index];
        p.PortName = data.NickName;
        if (data.HasDefaultValue)
        {
            p.UsingDefaultValue = true;
            p.DefaultValueEnabled = true;
       }
       p.ToolTipContent = data.ToolTipString;
       return p;
     }
    ...
}
``` 
If we do that through constructor, we have to copy some states from old `PortModel` to the new one (for example, `PortModel` maintains a list of connectors `ObservableCollection<ConnectorModel> Connectors`, we need to copy this list). Do you think adding this copy constructor is sort of overkill?